src/Mod/Assembly/JointObject.py

```diff
@@ -1494,19 +1494,35 @@ def adaptUi(self):
             self.form.distanceSpinbox.setProperty("minimum", float("-inf"))
             self.form.distanceSpinbox2.setProperty("minimum", float("-inf"))
 
-        if jType in JointUsingOffset:
-            self.form.offsetLabel.show()
-            self.form.offsetSpinbox.show()
-        else:
+        if self.form.advancedOffsetCheckbox.isChecked():
+            self.form.offset1Label.show()
+            self.form.offset2Label.show()
+            self.form.offset1Button.show()
+            self.form.offset2Button.show()
+
             self.form.offsetLabel.hide()
             self.form.offsetSpinbox.hide()
-
-        if jType in JointUsingRotation:
-            self.form.rotationLabel.show()
-            self.form.rotationSpinbox.show()
-        else:
             self.form.rotationLabel.hide()
             self.form.rotationSpinbox.hide()
+        else:
+            self.form.offset1Label.hide()
+            self.form.offset2Label.hide()
+            self.form.offset1Button.hide()
+            self.form.offset2Button.hide()
+
+            if jType in JointUsingOffset:
+                self.form.offsetLabel.show()
+                self.form.offsetSpinbox.show()
+            else:
+                self.form.offsetLabel.hide()
+                self.form.offsetSpinbox.hide()
```

Comment:
I'm not a fan of such approach, it requires you to remember both to show and hide the controls when necessary - which is an easy mistake to make. Instead, I'd introduce one boolean parameter per field and then call `setVisibility` on them. Something like:

```suggestion
            shouldShowOffset = jType in JointUsingOffset
            self.form.offsetLabel.setVisibility(shouldShowOffset)
            self.form.offsetSpinbox.setVisibility(shouldShowOffset)
```

You can do that for all controls and split state computation and actually updating the controls which then can be done on an not indented code path:

```
# compute the visibility of controls
shouldShowAdvancedOffsets = self.form.advancedOffsetCheckbox.isChecked()

shouldShowOffset   = jType in JointUsingOffset
shouldShowRotation = jType in JointUsingRotation

shouldShowOffset1 = shouldShowAdvancedOffsets
shouldShowOffset2 = shouldShowAdvancedOffsets

# update the visibility
self.form.offsetLabel.setVisibility(not shouldShowAdvancedOffsets and shouldShowOffset)
self.form.offsetSpinbox.setVisibility(not shouldShowAdvancedOffsets and shouldShowOffset)

...

Reply:
Yes you are correct using setVisibility is better. Though this small refactoring carry a risk of typo. This close to 1.0 I'm not sure if it's worth doing it.

Reply:
You can split it in two PRs. This would be for 1.1 and then refactor that will not be backported. Delaying such redactors is basically not doing them at all.